File: 202. Happy Number.cpp
class Solution {
public:
    int getExp(int no) {
        int sum = 0;
        while(no>0) {
            sum += (no%10)*(no%10);
            no/=10;
        }
        return sum;
    }
    bool isHappy(int n) {
        
        int exp = getExp(n);
        int i=0;
        while (1) {
            if (exp == 1) {
                return true;
            }
            exp = getExp(exp);
            i++;
            if (i>=100) {
                break;
            }
        }
        return false;
    }
};

File: 202. Happy Number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        exp = sum(map(lambda x:x*x, map(int, list(str(n)))))
        if exp == 1:
            return True
        i=0
        while True:
            exp = sum(map(lambda x:x*x, map(int, list(str(exp)))))
            if exp==1:
                return True
            i+=1
            if i>=100:
                break
            continue
        return False
